### PR TITLE
(maint) Enforce a License for all modules

### PR DESCRIPTION
**Pull request by glennsarti**
This commit adds a LICENSE file for all modules and automatically updates the
date when applied.  This commit also includes the new branding for Puppet.

### CODE REVIEW

**Comment by glennsarti** [comment]
I'm also expecting to raise another PR to merge this commit into master.

**Comment by jtappa** [comment]
Should the Appendix and how to paragraph on how to use the license remain in there? They seem like instructions for us, not users. Otherwise, :+1:

**Comment by glennsarti** [comment]
@jtappa Should I perhaps be using the license stub from Puppet itself?

https://raw.githubusercontent.com/puppetlabs/puppet/master/LICENSE

File: moduleroot/LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright <%= Date.today.year -%> Puppet, Inc.
```
Comment (ferventcoder) [inline moduleroot/LICENSE]: Interesting.

Comment (ferventcoder) [inline moduleroot/LICENSE]: Copyright usually lists all years from and to.

**Comment by ferventcoder** [comment]
@jtappa usually we include the Apache 2.0 License as it is stated in the LICENSE file - we also tend to include the NOTICE file which includes the attribute specific information. I just noticed that this PR mixes the two.

**Comment by glennsarti** [comment]
Ahh ok, so include the license verbatim but notice is Puppet specific?

**Comment by glennsarti** [comment]
There is no consistency about what to use :-(  e.g. pxp-agent vs puppetdb vs facter vs puppet-agent

**Comment by glennsarti** [comment]
Okay, so according to the Apache Foundation I should copy the license verbatim to LICENSE and add a NOTICE file

http://choosealicense.com/licenses/apache-2.0/
http://www.apache.org/licenses/LICENSE-2.0.html

**Comment by glennsarti** [comment]
Updated this PR with the following;
A verbatim copy of the Apache 2 license.
A NOTICE file that automatically adds in the module name, and pulls in copyright dates from modsync with a default of the current year.

There is some conjecture about what the dates should be.  TLDR version;
- The from date (optional) should not predate 'the work' being created.
- The to date (required) should be updated upon major changes to 'the work'.  It's up to the owner to decide what major is.

Note - As we've changed our legal name, all copyright notices need to be updated, which IMHO constitutes a major change and therefore date bump.

**Comment by ferventcoder** [comment]
Rebase please and thanks!

**Comment by ferventcoder** [comment]
> Okay, so according to the Apache Foundation I should copy the license verbatim to LICENSE and add a NOTICE file

Yep. That's what I was hoping for.

**Comment by ferventcoder** [comment]
> Note - As we've changed our legal name, all copyright notices need to be updated, which IMHO constitutes a major change and therefore date bump.

I think we should check with our legal on this one - I believe we will want something like

~~~md
Copyright 20xx-2016 Puppet Labs, Inc.
Copyright 2016 Puppet, Inc.
~~~

**Comment by glennsarti** [comment]
Aside from the whitespace, that's what I've got in the NOTICE

**Comment by glennsarti** [comment]
Rebased

File: config_defaults.yml [hunk 2]
```diff
@@ -44,6 +44,9 @@ appveyor.yml:
 MAINTAINERS.md:
   maintainers:
     - "Puppet Forge Modules Team `forge-modules |at| puppet |dot| com`"
+NOTICE:
+  # copyright-begin: '2011'   # Beginning date of copyright (*) Please consult legal as to what these dates ACTUALLY mean and when they should be changed
+  # copyright-end: '2015'     # End date of copyright (*) Please consult legal as to what these dates ACTUALLY mean and when they should be changed
```
Comment (ferventcoder) [inline config_defaults.yml]: This could be the year to date item :D

**Comment by ferventcoder** [comment]
This is a great start! Let's pull it in and find out how we need to change it later.

**Comment by glennsarti** [comment]
Normally I would agree but this PR will kick off a further 8-16 PRs for all of the Windows modules.  So I'd like to spend the time to get it right and then update.

**Comment by ferventcoder** [comment]
Makes sense - I'm just stating it may take a little more time to get some feedback from legal.

**Comment by glennsarti** [comment]
So let's timebox it.  Merge it of no response by COB 28th Apr.

**Comment by glennsarti** [comment]
Ok. TLDR from legal. Date range is preferred. From year of creation to latest change.
With that in mind.  I would like to merge this PR. Then update the modules with their dates of creation as maint PRs

**Comment by ferventcoder** [comment]
And what did they say about old name / new name?

File: config_defaults.yml (hunk 2 above)
Comment (ferventcoder) [inline config_defaults.yml]: This also doesn't handle multiple groups with copyright. I think you could go the simpler route of just making this text (and a list).

**Comment by glennsarti** [comment]
Oh right I see what you meant originally.  I've posed the question.  Waiting for response.

**Comment by glennsarti** [comment]
Legal have responded and a single entry is ok.  I've added support for multiple holders anyway.  Need to rebase still.

**Comment by glennsarti** [comment]
Rebased

**Comment by ferventcoder** [comment]
👍

**Pull request merged**